kasa/feature.py

```diff
@@ -92,10 +111,19 @@ def __post_init__(self):
             else:
                 self.category = Feature.Category.Info
 
+        if self.category == Feature.Category.Config and self.type in [
+            Feature.Type.Sensor,
+            Feature.Type.BinarySensor,
+        ]:
+            raise ValueError(
+                f"Invalid type for configurable feature: {self.name} ({self.id}):"
+                f" {self.type}"
+            )
+
```

Comment:
Shouldn't we something similar for `Feature.Category.Info` that is a switch or button etc?

Reply:
Hmm, so that original problem stems from the fact that we are defaulting to `Type.Sensor` if none is given while populating the category based on the existence of `attribute_setter`.

Perhaps the proper fix would be to default to `Type.Unknown` or `Type.Unset` to follow how categories work, and fall back to `Type.Sensor` if the feature defines only a getter?

Reply:
I want to get this merged to be able to add support for `Type.Choice` for the thermostats, so let's handle change to the logic in another PR as discussed on discord.